### PR TITLE
Adds voucher-only vending items + example + vendomat messages

### PR DESCRIPTION
**Pull request by SonixApache**
Adds a new framework for voucher-only vending machine items.
Also adds a glowing voucher that can be redeemed at vendomats for glowsticks, as an example.

Also adds and changes some messages regarding broken vendomats, to help the user figure out why it isn't working. Not 100% atomic, but I was in the file anyway.

Tested.

:cl:
 - rscadd: Added a framework for voucher-only vending machine items
 - tweak: Added some feedback messages for broken vending machines

### CODE REVIEW

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -1083,7 +1096,21 @@ var/global/num_vending_terminals = 1
...
+		/obj/item/device/assembly_frame = 1
+		)
+	vouched = list(
+		/obj/item/weapon/glowstick = 2,
+		/obj/item/weapon/glowstick/red = 2,
+		/obj/item/weapon/glowstick/blue = 2,
+		/obj/item/weapon/glowstick/yellow = 2,
+		/obj/item/weapon/glowstick/magenta = 2
+		)
+	prices = list( //Just for the glowing vouchers. Maybe should add a 'vouched' section to vendomats?
```
Comment (SonixApache) [inline code/game/machinery/vending.dm]: Wait, I DID add a 'vouched' section to vendomats.

**Comment by Blithering** [comment]
Vouchers are cool, it would be nice to have more ways to obtain them though, like Dr. Acula rewards or arcade prizes.

**Comment by SkowronX** [comment]
Buy them with pomf coins

**Comment by Blithering** [comment]
Voucher for icecream dispensers to give pomf brand icecream?

**Comment by RFultra** [comment]
GLOWING vouchers!

**Comment by Intigracy** [comment]
We could add continuity between rounds via rewarding successful antags with vouchers the next round they play.

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -360,7 +368,7 @@ var/global/num_vending_terminals = 1
 			src.update_vicon()
 			getFromPool(/obj/item/weapon/shard, loc)
 		else
-			to_chat(user, "<span class='notice'>[src] is broken! Fix it first.</span>")
+			to_chat(user, "<span class='notice'>The glass in the [src] is broken! Fix it first.</span>")
```
Comment (Intigracy) [inline code/game/machinery/vending.dm]: \the

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -621,6 +629,14 @@ var/global/num_vending_terminals = 1
 	damaged()
 
 /obj/machinery/vending/attack_hand(mob/living/user as mob)
+	if(stat & (BROKEN))
+		to_chat(user, "<span class='notice'>The glass in the [src] is broken, it refuses to work.</span>")
```
Comment (Intigracy) [inline code/game/machinery/vending.dm]: \the

File: code/game/machinery/vending.dm [hunk 4]
```diff
@@ -621,6 +629,14 @@ var/global/num_vending_terminals = 1
...
+		return
+	
+	if(stat & (NOPOWER))
+		to_chat(user, "<span class='notice'>The [src] is dark and unresponsive.</span>")
```
Comment (Intigracy) [inline code/game/machinery/vending.dm]: \the

**Review by Intigracy — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending.dm (hunk 4 above)
Comment (Intigracy) [reply code/game/machinery/vending.dm]: \The*

**Comment by SonixApache** [comment]
@Intigracy 
spongebob_the

\*edit*
image
ehhhh
all the vending machines are capitalised
should i just replace every instance of [src] with *the vending machine*?

**Comment by SonixApache** [comment]
image
seems aight now but i fucked up the CTRL+H hold on

**Comment by Exxion** [comment]
You don't need `\improper` before already-lowercase names
Also...

File: code/game/machinery/vending.dm [hunk 5]
```diff
@@ -1000,7 +1013,7 @@ var/global/num_vending_terminals = 1
 /*
 
 /obj/machinery/vending/[vendors name here]   // --vending machine template   :)
-	name = ""
+	name = "\improper "
```
Comment (Exxion) [inline code/game/machinery/vending.dm]: honk

Comment (SonixApache) [reply code/game/machinery/vending.dm]: I KNOW ;_;

**Comment by SonixApache** [comment]
yeah aight fixed that too

**Comment by Intigracy** [comment]
Add the voucher to the arcade maybe?

If more get added we can just make it a box like the other prizes.

**Comment by SonixApache** [comment]
There's only two (sets of) glowsticks per Vendomat, and only one or rarely two vendomats per map. If you manage to win more than two it becomes a wasted prize. I'd have to code a global limit on specific prizes for arcade machines and... blah. Could just add it to a vault if 100% necessary.

And honestly, the GLOWING voucher should be something you get from admins. Preferably ones from another server.

**Pull request merged**